tableschema/schema.py

```diff
@@ -132,19 +134,44 @@ def remove_field(self, name):
             self.__build()
         return field
 
-    def cast_row(self, row, fail_fast=False):
+    def cast_row(self, row, fail_fast=False, row_number=None,
```

Comment:
Schema.cast_row accepts an optional row_number for adding row number information in exception error message